src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMoveBookingService.kt

```diff
@@ -47,30 +53,63 @@ class PrisonPersonMoveBookingService(
 
   private suspend fun syncFromOffenderDps(fromOffenderNo: String, telemetry: MutableMap<String, String>) {
     physicalAttributesNomisApiService.getPhysicalAttributes(fromOffenderNo).also { nomisResponse ->
-      val latestBooking = nomisResponse.bookings.find { booking -> booking.latestBooking }
-      if (latestBooking != null) {
-        physicalAttributesSyncService.physicalAttributesChanged(fromOffenderNo, latestBooking.bookingId, nomisResponse)
-        telemetry += "syncFromOffenderDps_HEIGHT" to "true"
-        telemetry += "syncFromOffenderDps_WEIGHT" to "true"
-      }
+      nomisResponse.bookings
+        .find { it.latestBooking }
+        ?.also { latestBooking ->
+          physicalAttributesSyncService.physicalAttributesChanged(fromOffenderNo, latestBooking.bookingId, nomisResponse)
+          telemetry += "syncFromOffenderDps_HEIGHT" to "true"
+          telemetry += "syncFromOffenderDps_WEIGHT" to "true"
+        }
     }
 
-    // TODO SDIT-2200 sync all profile details for from offender
+    profileDetailsNomisApiService.getProfileDetails(fromOffenderNo).also { nomisResponse ->
+      nomisResponse.bookings
+        .find { it.latestBooking }
+        ?.also { latestBooking ->
+          latestBooking.profileDetails
+            .filter { it.type.isPhysicalAttributesProfileType() }
+            .forEach { profileDetail ->
+              profileDetailsSyncService.profileDetailsPhysicalAttributesChanged(profileDetail.type, fromOffenderNo, latestBooking.bookingId, nomisResponse)
+              telemetry += "syncFromOffenderDps_${profileDetail.type}" to "true"
+            }
+        }
+    }
   }
 
-  private suspend fun syncToOffenderDps(toOffenderNo: String, telemetry: MutableMap<String, String>) {
+  private suspend fun syncToOffenderDpsIfChanged(toOffenderNo: String, bookingId: Long, telemetry: MutableMap<String, String>) {
     physicalAttributesNomisApiService.getPhysicalAttributes(toOffenderNo).also { nomisResponse ->
-      val latestBooking = nomisResponse.bookings.find { booking -> booking.latestBooking }
-      if (latestBooking != null &&
-        latestBooking.findLastModifiedPhysicalAttributes().lastModifiedDateTime() > latestBooking.startDateTime
-      ) {
-        physicalAttributesSyncService.physicalAttributesChanged(toOffenderNo, latestBooking.bookingId, nomisResponse)
-        telemetry += "syncToOffenderDps_HEIGHT" to "true"
-        telemetry += "syncToOffenderDps_WEIGHT" to "true"
-      }
+      nomisResponse.bookings
+        .find { it.bookingId == bookingId }
+        ?.takeIf { it.latestBooking }
```

Comment:
Fix height/weight such that:
* only sync updates to NOMIS for the bookign that moved
* and only if it is the latest booking, not interested in historical booking moves